npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb

```diff
@@ -545,6 +492,22 @@ def updated_package_json_content(file)
             ).updated_package_json.content
         end
 
+        def prepared_package_json_content(content)
+          @prepared_package_json_content ||= {}
+          @prepared_package_json_content[content] ||=
+            package_json_preparer(
+              content
+            ).prepared_content
+        end
```

Comment:
Adding this as a memoized function seems extra compared to using `package_json_preparer(content).prepared_content` explicitly?

I don't feel strongly about this though so it is also fine to leave in.

Reply:
I removed the extra memoization and also found a similar one which I removed too at https://github.com/dependabot/dependabot-core/pull/7245/commits/81da5a1b40699053079e26730822daa94845160c.